scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -601,9 +613,10 @@
        (into {})))
 
 (defn scan-process
-  "Special verison of process run during scanning. It grabs the lock before processing the pod."
-  [{:keys [name] :as compute-cluster} pod-name]
-  (timers/time! (metrics/timer "scan-process" name)
-    (timers/time! (metrics/timer "process-lock" name)
-      (locking (calculate-lock pod-name)
-        (process compute-cluster pod-name)))))
+  "Given a pod-name, looks up the pod and delegates
+  to synthesize-state-and-process-pod-if-changed."
+  [{:keys [k8s-actual-state-map name] :as compute-cluster} pod-name]
+  (timers/time!
+    (metrics/timer "scan-process" name)
+    (let [{:keys [pod]} (get @k8s-actual-state-map pod-name)]
+      (synthesize-state-and-process-pod-if-changed compute-cluster pod))))
```

Comment:
## Why?

We need `scan-process` to re-synthesize the state so that time-dependent conditions (like the one being introduced in this PR) will get calculated even when no new message has come through on the pod watch.

Reply:
Added this info to the PR description